src/binary-reader-ir.cc

```diff
@@ -1110,6 +1114,28 @@ Result BinaryReaderIR::OnModuleName(string_view name) {
   return Result::Ok;
 }
 
+
+Result BinaryReaderIR::OnFunctionSymbol(Index index,
+                                        uint32_t flags,
+                                        string_view name,
+                                        Index function_index) {
+  if ((flags & WABT_SYMBOL_FLAG_UNDEFINED) == 0 ||
+      (flags & WABT_SYMBOL_FLAG_EXPLICIT_NAME) != 0) {
+
+    if (name.empty()) {
+      return Result::Ok;
+    }
+
+    Func* func = module_->funcs[function_index];
+    std::string dollar_name =
+      GetUniqueName(&module_->func_bindings, MakeDollarName(name));
```

Comment:
If this is forcing a `$` on the name, then that isn't great, `NameGenerator::GenerateName` is generally doing that optionally.

Reply:
NameGenerator::GenerateName  seems to have a lot more going on (as well as being private). Maybe I could add another function that checks for $ being already present and that's more streamlined to what wasm2wat (and wasm-decompile) needs?

Reply:
(so, it seems that not generating names with $ will be useful for wasm-decompile. I will take this into consideration)